app/src/main/java/com/pajato/android/gamechat/main/MainActivity.java

```diff
@@ -211,37 +165,13 @@ public boolean onNavigationItemSelected(MenuItem item) {
         return true;
     }
 
-    @Override public void onSaveInstanceState(final Bundle outState) {
-        super.onSaveInstanceState(outState);
-
-        // Store the Instructions and Turn for the save state.
-        outState.putStringArrayList(mKEY_INSTRUCTIONS, mInstructions);
-        outState.putBoolean(mKEY_TURN, mTurn);
-    }
-
     /**
-     * Sends a message alerting the event handling system that the new game button was clicked.
+     * An OnClick Listener for the new game button.
      *
      * @param view the new game button.
      */
     public void onNewGame(final View view) {
-        // Create the message.
-        String msg = getTurn() + "\n";
-        msg += view.getTag().toString();
-
-        // Empty the instructions list, as a new game has begun.
-        mInstructions.clear();
-        mInstructions.add(msg);
-
-        // Output New Game Message
-        String newTurn = (getTurn().equals(getString(R.string.player_1)) ?
-                "Player 1 (" + getString(R.string.xValue) + ")" :
-                "Player 2 (" + getString(R.string.oValue) + ")") + "'s Turn";
-        Snackbar start = Snackbar.make(findViewById(R.id.activity_main), "New Game! " + newTurn, Snackbar.LENGTH_SHORT);
-        start.show();
-
-        //TODO: replace this with an implemented event handling system.
-        sendMessage(msg);
+        ((TTTFragment) Panel.ttt.getFragment(getApplicationContext())).onNewGame(view);
```

Comment:
You would be wise to consider making the getFragment() method return something that extends Fragment so you can avoid the cast.